oaknational/Oak-Web-Application
# PUPIL-421, PUPIL-423: Enable worksheet downloads from the pupil intro

## Description

Music year: [1985](https://www.youtube.com/watch?v=GA4ROLQF4cs)

Enables the downloading of worksheets for new and legacy (🤞🏼) lessons

## Issue(s)
1. Though i've implemented this for both the new and legacy lessons there aren't any legacy lessons available to test with.
2. I've added a very basic error state should the download fail to start as a ([a simple alert](#failed-download)). If there are any examples of how to better handle this (or even a better message for the alert 😉 ) in the app elsewhere i'd be very happy to improve this.


## How to test

1. Go to https://deploy-preview-2237--oak-web-application.netlify.thenational.academy/pupils/programmes/combined-science-secondary-ks4-higher-aqa/units/measuring-waves/lessons/oscilloscope
3. Click on "Intro"
4. You should see the "Worksheet" card
5. Click "Download worksheet"
6. A zip file containing the worksheet should be downloaded 

## Screenshots

https://github.com/oaknational/Oak-Web-Application/assets/122096/e3641d8e-7a7c-49aa-8b08-584243ec107e


<span id="failed-download"></span>

## Failed download
https://github.com/oaknational/Oak-Web-Application/assets/122096/d1946e22-a0c5-4455-aad8-39e2dd5d8f96


## Checklist
